src/main/kotlin/domain/Board.kt

```diff
@@ -0,0 +1,5 @@
+package domain
+
+class Board(boardGenerator: BoardGenerator) {
+    val cells: Cells = boardGenerator.create()
+}
```

Comment:
cells가 더 중요한 데이터 아닐까요? `boardGenerator`가 주생성자에 있을 필요는 없어보입니다.

```suggestion
class Board(val cells: Cells) {
    constructor(boardGenerator: BoardGenerator) : this(boardGenerator.create())
}\
```